### ISSUE TITLE
introduce functions to work with spark

### ISSUE THREAD

**Issue by njtierney** [njtierney/naniar#40]
@MilesMcBain mentioned that it is difficult to work with missing data within spark.

For example:

```r
library(sparklyr)
library(tibble)
library(dplyr)

dat <- tribble(
    ~A, ~B, ~C,
    NA,  1,  1,
    1,  1, NA,
    NA, NA,  1,
    NA, NA, NA,
    1,  1,  1 
)

sc <- spark_connect(master = "local")
spark_dat <- copy_to(sc, dat)

#A crappy non-scalable way to do complete.cases
complete_cases <- 
    spark_dat %>% 
    filter(!is.na(A) & !is.na(B) & !is.na(C)) %>%
    collect()

#A crappy non-scalable way to do find rows with any na 
any_na <-
    spark_dat %>%
    filter(!(!is.na(A) & !is.na(B) & !is.na(C))) %>%
    collect()
```

It would be great to have naniar functions that also worked with spark. 

Not sure how much work this would involve, but it looks like rstudio have a pretty nice [extension API](http://spark.rstudio.com/extensions.html).

Just a thought for now, there's a lot of other things that I want to finish up first, but this should be on my roadmap towards version 1.0.0

**Comment by runkelcorey** [comment]
I see you haven't implemented any solutions, but have your ideas progressed on this since 2017? Would a fix be as easy as replacing dplyr / ggplot verbs with dbplyr and dbplot?

**Comment by njtierney** [comment]
I haven't yet! Things got in the way, and the spark/dbplyr implementation unfortunately didn't slide up to priority. 

I think it could be as simple as replacing dplyr with dbplyr, but then but it's not something I've explored as yet. One of the main issues is that I haven't actually run into needing to use dbplyr myself, and so I haven't had the urgent motivation to get this working, if that makes sense? 

I don't think `dbplot` will work as a drop in replacement as the ggplot objects in naniar are more custom than the ones offered in `dbplot`? But I guess they could be extended/customised from the initial objects they provide.

Hopefully I can get a bit more funding later in the year to focus on this!

**Comment by runkelcorey** [comment]
Yes, totally makes sense that it's not a priority! I find myself working in Databricks a lot these days, and could definitely see the use case for `naniar` (and `visdat`, FWIW) to operate on Spark-connected tibbles. I'd be happy to lend some time, but I wonder if you could size how heavy a lift this might be? My sense is that most functions will translate easily due to the expansive `sparklyr `API but a precious few low-level constructors or reshaping functions will house most of the work. If it helps, this guide could be relevant: [Creating Extensions for Sparklyr](https://spark.rstudio.com/guides/extensions.html). However, I haven't seen a guide aimed at package maintainers who want to convert already-built packages for use with Spark.
